### PR TITLE
[do not merge yet]Amax optimization

### PR DESCRIPTION
**Pull request by TonyYHsieh**
Amax optimization

### CODE REVIEW

File: library/include/hipblaslt-ext-op.h [hunk 1]
```diff
@@ -150,6 +150,12 @@ HIPBLASLT_EXPORT hipblasStatus_t hipblasltExtLayerNorm(hipDataType datatype,
  *  input 2-D tensor buffer. can't be nullptr.
  *
  *  @param[in]
+ *  workspace Amax tensor buffer (4k). can't be nullptr.
```
Comment (jichangjichang) [inline library/include/hipblaslt-ext-op.h]: Finding anchors and sections in /home/docs/checkouts/readthedocs.org/user_builds/advanced-micro-devices-hipblaslt/checkouts/752/library/include/hipblaslt-ext-op.h:138: error: argument 'workspace' of command @param is not found in the argument list of hipblasltExtAMax(const hipDataType datatype, const hipDataType outDatatype, void *output, void *input, void *workSpace, void *sync, uint32_t m, uint32_t n, hipStream_t stream) (warning treated as error, aborting now)
the documentation...

File: library/src/amd_detail/hipblaslt-ext-op.cpp [hunk 2]
```diff
@@ -511,27 +536,34 @@ hipblasStatus_t hipblasltAMaxWithScaleRun(const hipDataType datatype,
 
     const auto kernelName = sol->name();
     err                   = adapter->initKernel(kernelName);
+    int workSize          = 131072;
```
Comment (jichangjichang) [inline library/src/amd_detail/hipblaslt-ext-op.cpp:637]: hardcode??

Comment (TonyYHsieh) [reply library/src/amd_detail/hipblaslt-ext-op.cpp:637]: yes

Comment (TonyYHsieh) [reply library/src/amd_detail/hipblaslt-ext-op.cpp:637]: best value when we do experiment.

**Review by jichangjichang — COMMENTED** [review]
Please add new APIs for new cases. Old API should be kept for supporting backward compatibilities.

File: library/include/hipblaslt-ext-op.h (hunk 1 above)
Comment (TonyYHsieh) [reply library/include/hipblaslt-ext-op.h]: update

**Comment by TonyYHsieh** [comment]
> Please add new APIs for new cases. Old API should be kept for supporting backward compatibilities.

update

**Pull request closed without merging**